disco/extensions/upgrade_simulation/upgrades/common_functions.py

```diff
@@ -367,11 +365,81 @@ def get_upgrade_stage_stats(dss, upgrade_stage, upgrade_type, xfmr_loading_df, l
                                         "num_regulators": dss.RegControls.Count(),
                                         } )
+    
+    return final_dict
+
+def get_upgrade_stage_stats(dss, upgrade_stage, upgrade_type, xfmr_loading_df, line_loading_df, bus_voltages_df, **kwargs):
+    """This function gives upgrade stage stats for a feeder 
+    upgrade_stage can be Initial or Final
+    upgrade_type can be thermal or voltage
+    
+    """
+    final_dict = get_feeder_data(dss, upgrade_stage, upgrade_type)
     equipment_dict = combine_equipment_health_stats(xfmr_loading_df, line_loading_df, bus_voltages_df, **kwargs)
     final_dict.update(equipment_dict)
     return final_dict
 
 
+def get_upgrade_stage_timeseries_stats(dss, upgrade_stage, upgrade_type, **kwargs):
+    """This function gives upgrade stage stats for a feeder 
+    upgrade_stage can be Initial or Final
+    upgrade_type can be thermal or voltage
+    
+    """
+    final_dict = get_feeder_data(dss, upgrade_stage, upgrade_type)
+    equipment_dict = combine_equipment_health_timeseries_stats(**kwargs)
+    final_dict.update(equipment_dict)
+    return final_dict
+
+
+def combine_equipment_health_timeseries_stats(**kwargs):
+    line_properties = kwargs.pop("line_properties", 
+                                 ['name', 'phases','normamps', 'kV', 'line_placement',  'length', 'units', 'max_amp_loading', 
+                                  'max_per_unit_loading', 'status'])
+    xfmr_properties = kwargs.pop("xfmr_properties", 
+                                 ['name', 'phases', 'windings', 'conns', 'kV', 'kVA', 'amp_limit_per_phase','max_amp_loading', 
+                                  'max_per_unit_loading', 'status']  )
+    voltage_properties = kwargs.pop("voltage_properties", 
+                                    ['name', 'max_per_unit_voltage', 'min_per_unit_voltage',  'overvoltage_violation', 
+                                     'max_voltage_deviation', 'undervoltage_violation', 'min_voltage_deviation'])
+    capacitors_df = kwargs.pop("capacitors_df", pd.DataFrame())
+    regcontrols_df = kwargs.pop("regcontrols_df", pd.DataFrame())
+    capacitor_properties = kwargs.pop("capacitor_properties", 
+                                      ['capacitor_name','capcontrol_present',  'capcontrol_type', 'capcontrol_name', 'kv', 'kvar',
+                                       'phases', 'DeadTime', 'Delay', 'OFFsetting', 'ONsetting'])
+    regcontrol_properties = kwargs.pop("regcontrol_properties", 
+                                       ['name', 'transformer', 'vreg', 'band', 'ptratio', 'delay', 'at_substation_xfmr_flag'])
+    
+    final_dict = {}
+    xfmr_dict = get_timeseries_thermal_comparison(equipment_type="transformer", compute_loading=True, upper_limit=kwargs["transformer_upper_limit"])
+    ref_xfmr_dict = {k:v.reset_index()[xfmr_properties].to_dict(orient="records") for k, v in xfmr_dict.items()}

```

Comment:
Why do you need to reset the index in all of these dataframes?

Reply:
This is cause the dataframe received has name as index